### PR TITLE
feat(controller): Deprecate X prefixed headers

### LINKED ISSUE #2667: X- prefix is deprecated in headers
@philips pointed this out to me earlier today: https://tools.ietf.org/html/rfc6648

We should take a look at the headers we use (X_DEIS_RELEASE, X_DEIS_PLATFORM_VERSION, X_DEIS_API_VERSION to name a few) and deprecate them in favour of DEIS_RELEASE etc.

### PR DESCRIPTION
**Pull request by Joshua-Anderson**
`X_DEIS_API_VERSION` ->  `DEIS_API_VERSION`
`X_DEIS_PLATFORM_VERSION` -> `DEIS_PLATFORM_VERSION`
`X-Deis-Release` -> `Deis-Release`
Since this is a important change I bumped the api version.

*edit: carmstrong*
closes #2667

### CODE REVIEW

**Comment by bacongobbler** [comment]
Perfectly executed. Code LGTM!

**Comment by bacongobbler** [comment]
closes #2667

**Comment by bacongobbler** [comment]
Would you also mind deprecating https://github.com/deis/deis/blob/master/controller/api/views.py#L102? Thanks!

**Comment by Joshua-Anderson** [comment]
@bacongobbler Done!

File: controller/api/middleware.py [hunk 1]
```diff
@@ -46,5 +46,6 @@ def process_response(self, request, response):
         a response header.
         """
         # clients shouldn't care about the patch release
-        response['X_DEIS_API_VERSION'] = __version__.rsplit('.', 1)[0]
+        response['DEIS_API_VERSION'] = __version__.rsplit('.', 1)[0]
+        response['X_DEIS_API_VERSION'] = __version__.rsplit('.', 1)[0]  # DEPRECATED
```
Comment (mboersma) [inline controller/api/middleware.py]: Maybe have the old header assign its value from the new one, instead of finding the `rsplit` twice?

**Comment by mboersma** [comment]
Code LGTM, my nitpick is optional.

**Comment by Joshua-Anderson** [comment]
This test failed for a really weird reason. `auth:login http://deis.test-06a0bac476.deis.works --username=test1 --password=test1` failed with:

```
Couldn't connect to the Deis Controller:
('Connection aborted.', BadStatusLine("''",))
Make sure that the Controller URI is correct and the server is running.
```

However, `deis login` worked many times before this in the build, causing me to think this is a sporadic error. Could somebody rerun this CI build? Any ideas why this is happening?

**Comment by mboersma** [comment]
> Any ideas why this is happening?

Not exactly. We've seen this a few times. I'll try to track it down. It's unrelated to your PR, which passed when I ran it again.

Code LGTM.

**Pull request merged**